### PR TITLE
tidb: update docs for new transaction features in 5.0

### PR DESCRIPTION
**Pull request by sticnarf**
### What is changed, added or deleted? (Required)

Synchronize docs with latest changes in transaction features like async commit and 1PC.
 
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

~~Don't merge this PR until https://github.com/pingcap/tidb/pull/20313 is merged.~~

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

**Comment by sticnarf** [comment]
PTAL @TomShawn @cfzjywxk

File: tidb-configuration-file.md [hunk 1]
```diff
@@ -441,11 +441,38 @@ prepare 语句的 plan cache 设置。
...
++ 指定是否在只涉及一个 region 的事务上使用一阶段提交特性。比起传统两阶段提交，一阶段提交能大幅提升事务提交的延迟和吞吐表现。
++ 默认值：false
+
+> **警告：**
+>
+> 当前该功能为实验特性，不建议在生产环境中使用。目前存在已知问题有：
+>
+> + 暂时与 [TiCDC](/ticdc/ticdc-overview.md) 不兼容，可能导致 TiCDC 运行不正常。
+> + 暂时与 [Follower Read](/follower-read.md) 及 [TiFlash](/tiflash/tiflash-overview.md) 不兼容，使用时无法保证快照隔离。
+> + 无法保证外部一致性。
```
[the remark this answers is not in the document]
Comment (sticnarf) [reply tidb-configuration-file.md:455]: 只要 Commit TS 是计算出来的，都可能有。举个例子：
- 在同一个 session 中，在 TiKV 1 上提交 k1@20，提交成功后在 TiKV 2 上提交 k2@10，这两个事务都是 1PC
- 另一个事务用 ts=15 读取，读到 k2 的变更但没读到 k1 的变更。
这样就同时出现了 k1 < k2 和 k1 > k2 两个情况，也不满足外部一致性。

[the remark this answers is not in the document]
Comment (cfzjywxk) [reply tidb-configuration-file.md:450]: Got, thx~

**Comment by cfzjywxk** [comment]
LGTM

File: tidb-configuration-file.md [hunk 2]
```diff
@@ -436,11 +436,38 @@ prepare 语句的 plan cache 设置。
 + TiKV 的负载阈值，如果超过此阈值，会收集更多的 batch 封包，来减轻 TiKV 的压力。仅在 `tikv-client.max-batch-size` 值大于 0 时有效，不推荐修改该值。
 + 默认值：200
 
-### `enable-async-commit` <!-- 从 v5.0 版本开始引入 -->
+### `enable-one-pc` <!-- 从 v5.0 版本开始引入 -->
+
++ 指定是否在只涉及一个 region 的事务上使用一阶段提交特性。比起传统两阶段提交，一阶段提交能大幅降低事务提交延迟并提升吞吐。
```
Comment (TomShawn) [inline tidb-configuration-file.md:441]:
```suggestion
+ 指定是否在只涉及一个 Region 的事务上使用一阶段提交特性。比起传统两阶段提交，一阶段提交能大幅降低事务提交延迟并提升吞吐。
```

File: tidb-configuration-file.md [hunk 3]
```diff
@@ -436,11 +436,38 @@ prepare 语句的 plan cache 设置。
...
+> **警告：**
+>
+> 当前该功能为实验特性，不建议在生产环境中使用。目前存在已知问题有：
+>
+> + 暂时与 [TiCDC](/ticdc/ticdc-overview.md) 不兼容，可能导致 TiCDC 运行不正常。
+> + 暂时与 [Follower Read](/follower-read.md) 及 [TiFlash](/tiflash/tiflash-overview.md) 不兼容，使用时无法保证快照隔离。
+> + 无法保证外部一致性。
+> + 如果在执行 DDL 操作的同时，由于 TiDB 机器宕机等原因导致事务提交异常中断，可能造成数据格式不正确。
+
+## tikv-client.async-commit <!-- 从 v5.0 版本开始引入 -->
```
Comment (TomShawn) [inline tidb-configuration-file.md:453]:
```suggestion
## `tikv-client.async-commit` <!-- 从 v5.0 版本开始引入 -->
```

Comment (sticnarf) [reply tidb-configuration-file.md:453]: 我参考了 tikv-client.copr-cache 的格式，是没有代码格式的

File: tidb-configuration-file.md [hunk 4]
```diff
@@ -436,11 +436,38 @@ prepare 语句的 plan cache 设置。
...
+
+### `enable`
 
 + 指定是否启用 async commit 特性，使事务两阶段提交的第二阶段于后台异步进行。开启本特性能降低事务提交的延迟。本特性与 [TiDB Binlog](/tidb-binlog/tidb-binlog-overview.md) 不兼容，开启 binlog 时本配置将没有效果。
 + 默认值：false
 
+### `keys-limit`
+
++ 指定一个 async commit 事务中键的数量上限。过大的事务不适合使用 async commit，超出该限制的事务会使用传统两阶段提交方式。
```
Comment (TomShawn) [inline tidb-configuration-file.md:462]:
```suggestion
+ 指定一个 Async Commit 事务中键的数量上限。过大的事务不适合使用 Async Commit，超出该限制的事务会使用传统两阶段提交方式。
```

File: tidb-configuration-file.md [hunk 5]
```diff
@@ -436,11 +436,38 @@ prepare 语句的 plan cache 设置。
...
++ 默认值：256
+
+### `total-key-size-limit`
+
++ 指定一个 async commit 事务中键的大小总和的上限。如果事务涉及的键过长，则不适合使用 async commit，超出该限制的事务会使用传统两阶段提交方式。
```
Comment (TomShawn) [inline tidb-configuration-file.md:467]:
```suggestion
+ 指定一个 Async Commit 事务中键的大小总和的上限。如果事务涉及的键过长，则不适合使用 Async Commit，超出该限制的事务会使用传统两阶段提交方式。
```

**Review by TomShawn — COMMENTED** [review]
minor comments on format

**Comment by TomShawn** [comment]
LGTM

**Comment by sticnarf** [comment]
@TomShawn  https://github.com/pingcap/tidb/pull/20313 has been merged. Could you merge this PR?

**Review by TomShawn — APPROVED** [review]

**Pull request merged**